shell/platform/android/io/flutter/embedding/engine/plugins/lifecycle/HiddenLifecycleReference.java

```diff
@@ -0,0 +1,31 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.embedding.engine.plugins.lifecycle;
+
+import android.arch.lifecycle.Lifecycle;
+import android.support.annotation.NonNull;
+
+/**
+ * An {@code Object} that can be used to obtain a {@link Lifecycle} reference.
+ * <p>
+ * <strong>DO NOT INSTANTIATE THIS CLASS IN AN APP.</strong>
+ * <p>
+ * This class is used by the flutter_android_lifecycle package to provide access to a
+ * {@link Lifecycle} in a way that makes it easier for Flutter and the Flutter plugin ecosystem to
+ * handle breaking changes in Lifecycle libraries.
+ */
+public class HiddenLifecycleReference {
```

Comment:
I'm not sure, but I'd recommend against it. I think the answer to that question will always be determined by the details of Dalvik, Art, and any future bytecode system. If this were Objective-C then we might be able to do what you're saying because those dispatch tables can use strings as method identifiers, but the closest thing that Java has is reflection, which I think would be more trouble than it's worth.